prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -278,17 +281,22 @@ class Hl7Serializer(val metadata: Metadata) {
         report.schema.elements.forEach { element ->
             val value = report.getString(row, element.name) ?: return@forEach
 
+            if (suppressedFields.contains(element.hl7Field))
+                return@forEach
+
             if (element.hl7OutputFields != null) {
-                element.hl7OutputFields.forEach { hl7Field ->
+                element.hl7OutputFields.forEach outputFields@{ hl7Field ->
+                    if (suppressedFields.contains(hl7Field))
+                        return@outputFields
```

Comment:
If we are not outputting to this field because we're suppressing it's value then we want to skip setting it.